bftengine/src/bftengine/SimpleClientImp.cpp

```diff
@@ -446,11 +453,11 @@ uint64_t SeqNumberGeneratorForClientRequestsImp::generateUniqueSequenceNumberFor
       LOG_WARN(GL, "Client SeqNum Counter reached max value");
       lastMilliOfUniqueFetchID_++;
       lastCountOfUniqueFetchID_ = 0;
-    } else {
+    } else {  // increase last count to preserve uniqueness.
       lastCountOfUniqueFetchID_++;
```

Comment:
I suppose the `else` happens when the system clock "goes back in time", as pointed out by Andrew previously. Even if this happens, incrementing `lastCountOfUniqueFetchID_` guarantees the uniqueness of the generated SN.

If not already done, can you please add a unit test for this case?